### PR TITLE
Remove usages of `org.apache.commons.io.output.ByteArrayOutputStream`

### PR DESCRIPTION
**Pull request by basil**
I have no idea why `org.apache.commons.io.output.ByteArrayOutputStream` exists to begin with: its Javadoc states

> Implements a ThreadSafe version of [AbstractByteArrayOutputStream](https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/output/AbstractByteArrayOutputStream.html) using instance synchronization.

perhaps implying that `java.io.ByteArrayOutputStream` might not be thread-safe, but I checked the implementation in OpenJDK and it has been thread-safe since the first commit in 2007. It seems perfectly safe to replace this with the native Java Platform class, and doing so seems desirable in the sense that it decreases our reliance on this third-party library. Doing so woke up SpotBugs with a few `DM_DEFAULT_ENCODING` warnings, which I added to the "needs triage" list to eliminate any risk from this PR.

### Testing done

`mvn clean verify -Dtest=hudson.console.AnnotatedLargeTextTest,hudson.console.ConsoleAnnotatorTest,hudson.FilePathTest,hudson.model.UsageStatisticsTest,jenkins.slaves.OldRemotingAgentTest`

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8146"></a>

### CODE REVIEW

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by timja** [comment]
Isn't this the javadoc:
https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/output/ByteArrayOutputStream.html

Which gives this reason:
> This is an alternative implementation of the [ByteArrayOutputStream](http://docs.oracle.com/javase/7/docs/api/java/io/ByteArrayOutputStream.html?is-external=true) class. The original implementation only allocates 32 bytes at the beginning. As this class is designed for heavy duty it starts at 1024 bytes. In contrast to the original it doesn't reallocate the whole memory block but allocates additional buffers. This way no buffers need to be garbage collected and the contents don't have to be copied to the new buffer. This class is designed to behave exactly like the original. The only exception is the deprecated toString(int) method that has been ignored.

**Comment by basil** [comment]
Thanks for digging up the original motivation. This all seems like premature optimization to me, and I do not think we are losing anything important by switching to the Java Platform's version.

**Review by NotMyFault — APPROVED** [review]

**Pull request merged**
